test/TestUtilities/SystemUtil.hpp

```diff
@@ -54,8 +54,8 @@ struct ApproxVec
 // Utilities for approximate equality
 struct ApproxDeg
 {
-    ApproxDeg(double degrees)
-        : degrees(static_cast<float>(degrees)) {}
+    ApproxDeg(double deg)
+        : degrees(static_cast<float>(deg)) {}
```

Comment:
This is an annoying warning, which also forced us to do ugly stuff like `thePosition` in other places. It's even worse when the parameter name appears in the public API and thus the doc.

I guess it's not possible to selectively disable it only for constructors?

Reply:
`-Wshadow` does get very annoying when writing constructors for structs. It pushes me to use aggregate initialization for structs because of this problem. I don't believe it can be selectively disabled for constructors.

Reply:
What I found: 
* clang: https://reviews.llvm.org/D18271?id=55044
* gcc: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=78147

> Clang has an extra warning flag, -Wshadow-field-in-constructor, which controls this (from what I can tell).  By default in Clang, -Wshadow doesn't enable this you need to separately enable it.

Doesn't look like g++ has adopted anything similar.

Reply:
As we're not enabling any new warnings here, lets continue the shadow discussion separately.